udata/harvest/backends/base.py

```diff
@@ -136,6 +144,13 @@ def has_feature(self, key):
     def get_filters(self):
         return self.config.get("filters", [])
 
+    def get_extra_config_value(self, key: str):
+        try:
+            extra_config = next(c for c in self.config.get("extra_configs", []) if c["key"] == key)
+            return extra_config["value"]
+        except StopIteration:
+            return None
+
```

Comment:
Is this better than
```python
extra_configs = (c for c in self.config.get("extra_configs", []) if c["key"] == key])
if extra_configs:
    return extra_configs[0]["value"]
```

 or as written below in [udata/harvest/forms.py](https://github.com/opendatateam/udata/pull/3157/files#diff-b07a8dd68c58d7567e07b4f46a1eb6963d632c2021108980b177ce9ee7aa056cR27-R29)

```python
candidates = (c for c in self.config.get("extra_configs", []) if c["key"] == key])
return next(candidates, {"value": None})["value"]
```
?

Reply:
You're right, that was a dumb copy paste from `has_feature`. Changed it in https://github.com/opendatateam/udata/pull/3157/commits/515dba1a976cda14567fd47939ef74f97715f5ac.